src/bin/pg_autoctl/lock_utils.c

```diff
@@ -194,14 +193,21 @@ semaphore_cleanup(const char *pidfile)
 
 	lineCount = splitLines(fileContents, fileLines, BUFSIZE);
 
-	if (lineCount < 2)
+	if (lineCount < PIDFILE_LINE_SEM_ID)
 	{
+		log_debug("Failed to cleanup the semaphore from stale pid file \"%s\": "
+				  "it contains %d lines, semaphore id is expected in line %d",
+				  pidfile,
+				  lineCount,
+				  PIDFILE_LINE_SEM_ID);
+		free(fileContents);
 		return false;
 	}
 
-	if (!stringToInt(semIdString, &(semaphore.semId)))
+	if (!stringToInt(fileLines[PIDFILE_LINE_SEM_ID], &(semaphore.semId)))
 	{
 		/* errors have already been logged */
+		free(fileContents);
 		return false;
 	}
```

Comment:
I think you missed one here in the happy path:
```suggestion
	}
	free(fileContents);
```

Reply:
Done now, thanks.